### DISCUSSION TITLE
The model has an exponential term that reports an error

### DISCUSSION THREAD

**Discussion by Devil-li-x** [casadi/casadi#3162] [Q&A]
Hello Hello,
Recently I used Casadi in Python to do a simple model predictive control.The model part looks like this
`def ode(x,u):
    theta = 20
    k = 300
    M = 5
    xf = 0.3947
    xc = 0.3816
    alpha = 0.117
    c = x[0]
    T = x[1]
    rate = k*c*exp(M/T)
    dcdt = (1-c)/theta-rate
    dTdt = (xf-T)/theta-alpha*u*(T-xc)+rate
    return np.array([dcdt,dTdt])`
Error message：    `CasADi -  WARNING("ControlSolver:nlp_jac_g failed: NaN detected for output jac_g_x, at nonzero index 18 (row 0, col 3).") [.../casadi/core/oracle_function.cpp:265]
CasADi -  WARNING("ControlSolver:nlp_jac_g failed: NaN detected for output jac_g_x, at nonzero index 18 (row 0, col 3).") [.../casadi/core/oracle_function.cpp:26`
My constraints are all constants.
Once I delete the exponential term, there is no such error message, I don't know what caused it.

**Comment by brunoml5** [comment]
Hi! There are a lot of divisions in your code. Make sure that values close to zero in the denominator are not possible.

**Comment by jgillis** [comment]
There is a relevant FAQ entry: https://github.com/casadi/casadi/wiki/FAQ:-Why-am-I-getting-%22NaN-detected%22in-my-optimization%3F
